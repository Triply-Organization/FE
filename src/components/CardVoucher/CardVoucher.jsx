import { Button } from 'antd';
import PropTypes from 'prop-types';
import React from 'react';
import { BsArrowRight } from 'react-icons/bs';

import './CardVoucher.scss';

const CardVoucher = props => {
  const {
    title,
    subTitle,
    buttonContent,
    buttonOnClick,
    background,
    btnBgColor,
    ...prop
  } = props;
  return (
    <div className="card-voucher" {...prop}>
      <div className="card-voucher__content">
        <div className="card-voucher__content__typho">
          <h2>{title}</h2>
          <p>{subTitle}</p>
        </div>
        <Button
          className="card-voucher__content__button"
          type="primary"
          style={{ backgroundColor: btnBgColor }}
          onClick={buttonOnClick}
        >
          {buttonContent}
          <BsArrowRight className="icon" />
        </Button>
      </div>

      <img src={background} alt={title} />
    </div>
  );
};

CardVoucher.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
  buttonContent: PropTypes.string,
  buttonOnClick: PropTypes.func,
  background: PropTypes.string,
  btnBgColor: PropTypes.string,
};

export default CardVoucher;
