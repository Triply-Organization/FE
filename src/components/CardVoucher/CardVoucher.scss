.card-voucher {
  position: relative;
  border-radius: 8px;
  overflow: hidden;

  &__content {
    width: 60%;
    display: flex;
    flex-direction: column;
    gap: 1rem;
    position: absolute;
    top: 2rem;
    left: 2rem;
    z-index: 2;

    &__typho {
      width: 100%;
      & h2 {
        width: 80%;
        color: white;
        font-weight: bold;
        text-shadow: 0 1px 5px rgb(0 0 0 / 30%);
        font-size: 30px;
        line-height: 30px;
      }
      & p {
        font-weight: bold;
      }
    }
    &__button {
      width: 150px;
      height: 50px;
      display: flex;
      justify-content: center;
      gap: 1rem;
      align-items: center;
      border-radius: 6px;
      font-weight: bold;

      & > .icon {
        font-weight: bold;
        font-size: 1rem;
      }
    }
  }

  & > img {
    border-radius: 8px;
    transition: ease 300ms;
  }

  &:hover {
    & > img {
      transform: scale(1.2);
    }
  }
}
