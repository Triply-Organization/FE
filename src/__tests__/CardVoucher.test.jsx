import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { describe, expect, test } from 'vitest';

import CardVoucher from '../components/CardVoucher/CardVoucher';

describe('component/CardVoucher', () => {
  test('render correctly', () => {
    expect(
      render(
        <BrowserRouter>
          <CardVoucher />
        </BrowserRouter>,
      ),
    ).toMatchSnapshot();
  });
});
