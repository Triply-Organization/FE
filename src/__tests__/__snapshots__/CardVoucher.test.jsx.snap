// Vitest Snapshot v1

exports[`component/CardVoucher > render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-voucher"
      >
        <div
          class="card-voucher__content"
        >
          <div
            class="card-voucher__content__typho"
          >
            <h2 />
            <p />
          </div>
          <button
            class="ant-btn ant-btn-primary card-voucher__content__button"
            type="button"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <img />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-voucher"
    >
      <div
        class="card-voucher__content"
      >
        <div
          class="card-voucher__content__typho"
        >
          <h2 />
          <p />
        </div>
        <button
          class="ant-btn ant-btn-primary card-voucher__content__button"
          type="button"
        >
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <img />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
